test/bin/pyutils/common.py

```diff
@@ -43,22 +43,31 @@ def close_junit():
     JUNIT_LOGFILE = None
 
 
-def record_junit(object, step, status):
+def record_junit(object, step, status, start=0.0, log_filepath=''):
     """Add a message for the specified object and step with OK, SKIP or FAIL status.
     Recording messages is synchronized and it can be called from different threads.
     """
+    t = ''
+    if start != 0.0:
+        duration = time.time() - start
+        t = f' time="{duration}"'
+
     try:
         # BEGIN CRITICAL SECTION
         JUNIT_LOCK.acquire()
 
-        append_file(JUNIT_LOGFILE, f'<testcase classname="{object}" name="{step}">')
+        append_file(JUNIT_LOGFILE, f'<testcase classname="{object}" name="{step}"{t}>')
         # Add a message according to the status
         if status == "OK":
             pass
         elif status.startswith("SKIP"):
             append_file(JUNIT_LOGFILE, f'<skipped message="{status}" type="{step}-skipped" />')
         elif status.startswith("FAIL"):
-            append_file(JUNIT_LOGFILE, f'<failure message="{status}" type="${step}-failure" />')
+            desc = ''
+            if log_filepath:
+                with open(log_filepath) as f:
+                    desc = f.read()
+            append_file(JUNIT_LOGFILE, f'<failure message="{status}" type="${step}-failure">\n{desc}\n</failure>')
```

Comment:
Log files can be rather large. Do you see a problem with that?

Reply:
I changed to get only last 20 lines of the log. PTAL at the result with the debug commit: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_microshift/4211/pull-ci-openshift-microshift-main-e2e-aws-tests-bootc/1859888074561949696